dice.h

```diff
@@ -0,0 +1,20 @@
+#include <iostream>
+#include <random>
+#include <regex>
+#include <vector>
+#include <string>
```

Comment:
It looks like some of these could go in the CPP file instead. That will improve build time (which is a big reason for having header files).